### PR TITLE
[Fixes #296] Delete observation button for admin when no data

### LINKED ISSUE #296: Add button to delete observation
On observation single page, there should be a button to delete the observation visible and actionable only when:

* Viewed by admins
* Observation contains only empty data from ground stations.

### PR DESCRIPTION
**Pull request by faulteh**
Does some basic checks on the existences of payloads and if the user is staff also presents a 'Delete Observation' button.

### CODE REVIEW

File: network/base/views.py [hunk 1]
```diff
@@ -384,17 +382,31 @@ def prediction_windows(request, sat_id, start_date, end_date, station_id=None):
...
+    if request.user.is_staff:
+        data_payload_exists = False
+        for data in dataset:
+            if data.payload_exists:
+                data_payload_exists = True
+    # This context flag will determine if a delete button appears for the observation.
+    is_deletable = False
+    if observation.author == request.user and observation.is_deletable:
+        is_deletable = True
+    if request.user.is_staff and not data_payload_exists:
```
Comment (comzeradd) [inline network/base/views.py]: Wouldn't that give delete permissions to all staff, regardless if the have admin rights or not?

Comment (faulteh) [inline network/base/views.py]: Fixed in latest push now uses 'base.delete_observation' permission.

**Comment by comzeradd** [comment]
Tested locally and it works great.

One minor consideration is that a staff with permissions can delete an observation right away and that will result the client trying to push to a non-existing observation. For observers we solved this by having a short timespan for deletion, so the client won't have the time to fetch and schedule that specific observation. I'm thinking we should add a similar constraint here. Something like `OBSERVATION_MIN_DELETION_RANGE`, that will timedelta from the end of the observation window. So we could easily say that a staff can delete an observation only if 1h for example has passed since the observation window closed.

Does this makes sense @faulteh ?

**Comment by faulteh** [comment]
so the obs model now has a is_deletable_before_start and is_deletable_after_end properties

**Pull request merged**